src/app/header/header.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-header',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss']
+})
+export class HeaderComponent implements OnInit {
+
+  constructor(private router: Router) {}
+
+  showDropdown: boolean = false;
+
+  private _searchDetails: string = '';
+
+  get searchDetails(): string {
+    return this._searchDetails;
+  }
+
+  set searchDetails(value: string) {
+    this._searchDetails = value;
+    console.log('search: ', this._searchDetails)
+  }
+
+  toggleDropdown(): void {
+    this.showDropdown = !this.showDropdown;
+    console.log(this.showDropdown)
+  }
```

Comment:
See comments on categories and console.log

Reply:
I have removed categories and all console.logs